src/main/java/randoop/main/CoverageTracker.java

```diff
@@ -0,0 +1,192 @@
+
+  /**
+   * Coverage details related to a single method under test. Tracks total number of branches and
+   * number of uncovered branches.
+   */
+  public static class CoverageDetails {
+    private int numBranches;
+    private int uncoveredBranches;
+
+    private void setNumBranches(int numBranches) {
+      this.numBranches = numBranches;
+    }
+
+    private void setUncoveredBranches(int uncoveredBranches) {
+      this.uncoveredBranches = uncoveredBranches;
+    }
+
+    public int getNumBranches() {
+      return numBranches;
+    }
+
+    public int getUncoveredBranches() {
+      return uncoveredBranches;
+    }
+  }
+
+  /** Initializes the coverage tracker. */
+  private CoverageTracker() {
+    instrumenter = new Instrumenter(runtime);
+
+    data = new RuntimeData();
+    try {
+      runtime.startup(data);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Instruments and loads into memory, each class that is under test.
+   *
+   * @param classNames {@code Set} containing the names of the classes under test.
+   */
+  public void instrumentAndLoadAllClasses(Set<String> classNames) {
+    if (classNames == null) {
+      return;
+    }
+
+    // Instrument and load into memory, each class that is under test.
+    for (String className : classNames) {
+      final byte[] instrumented;
+
+      final String resource = '/' + className.replace('.', '/') + ".class";
+      InputStream original = getClass().getResourceAsStream(resource);
+
+      if (original == null) {
+        System.err.println("No resource with name: " + resource + " found!");
+        continue;
+      }
+
+      try {
+        // Instrument the class to prepare for coverage collection later.
+        instrumented = instrumenter.instrument(original, className);
+        original.close();
+      } catch (IOException e) {
+        e.printStackTrace();
+        continue;
+      }
+
+      memoryClassLoader.addDefinition(className, instrumented);
+    }
+  }
+
+  /**
+   * Returns the instrumented version of the class.
+   *
+   * @param className name of the class
+   * @return {@Code Class} object that has been instrumented for coverage data collection. null if
+   *     class with target name cannot be found.
+   */
+  public Class<?> getInstrumentedClass(String className) {
+    if (className == null || className.isEmpty()) {
+      return null;
+    }
+
+    // Check our cache first, to see if this class has already been loaded.
+    Class<?> instrumentedClass = instrumentedClasses.get(className);
+    if (instrumentedClass != null) {
+      return instrumentedClass;
+    }
+
+    try {
+      instrumentedClass = memoryClassLoader.loadClass(className);
+      instrumentedClasses.put(className, instrumentedClass);
+    } catch (ClassNotFoundException e) {
+      e.printStackTrace();
+    }
+    return instrumentedClass;
+  }
+
+  /** Collect coverage information for all methods under test. Updates the coverageDetailsMap. */
```

Comment:
Clarify whether this generates data (maybe by executing tests), or summarizes already-collected data, or prepares for later collection, or something else.